File: crawler/scala/src/test/scala/crawler/StockDiscussionCrawlerTest.scala
package crawler

import org.scalatest.FunSuite

class StockDiscussionCrawlerTest extends FunSuite {
    test("test get discussion") {
      val crawler = new StockDiscussionCrawler("005930")
      val discussion = crawler.getDiscussion("/item/board_read.nhn?code=005930&nid=163402849&st=&sw=&page=8")

      assert(discussion.title.equals("삼성전자 단기7만  중장기6만이에요."))
      assert(discussion.content.equals("지금 늦기전에 매수하고 7만까지 버티고 6만가면 매도 하세요."))
      assert(discussion.nextDiscussionUrl.startsWith("board_read.nhn?code=005930&nid=163402840"))
      assert(discussion.previousDiscussionUrl.startsWith("board_read.nhn?code=005930&nid=163402943"))

    }

  test("test no previous discussion") {
    val crawler = new StockDiscussionCrawler("005930")
    val firstUrl = crawler.getStartDiscussionUrl

    val discussion = crawler.getDiscussion(firstUrl)

    print(discussion)
    assert(discussion.previousDiscussionUrl.length < 1)
    assert(discussion.nextDiscussionUrl.length > 10)

  }

  test("test no next discussion") {
    val crawler = new StockDiscussionCrawler("005930")
    val discussion = crawler.getDiscussion("/item/board_read.nhn?code=005930&nid=64104583&st=&sw=&page=46711")

    print(discussion)
    assert(discussion.nextDiscussionUrl.length < 1)
    assert(discussion.previousDiscussionUrl.length > 10)
  }

}

File: src/test/kotlin/com/stock/crawler/StockDiscussionCrawlerTest.kt
package com.stock.crawler


import org.junit.jupiter.api.Test

class StockDiscussionCrawlerTest {

    @Test
    fun testGetDiscussion() {
        val crawler = NaverStockDiscussionCrawler("005930", 1)
        val discussion = crawler.getDiscussion("/item/board_read.nhn?code=005930&nid=64102696&st=&sw=&page=46750")

        println(discussion)
        assert(discussion.title.equals("문가가  정말 그랬을까에  자서전 당장 사봐야겠네에  댕큐"))
        assert(discussion.date.equals(""))
        assert(discussion.content.equals("베트남이 패망한것을 보며 희열을 느꼈다고 자서전에 고백했듯이 625에 김일성이 수백만 명을 죽인 것에 지금도 희열을 느껴 조기를 달지 못하게 했느냐 조기 없는 현충일은 처음본다 625 남침을 대통령이 김정은과 한패가 되어 애국이라고 떠벌이니"))
        assert(discussion.nextDiscussionUrl.startsWith("board_read.nhn?code=005930&nid=64102679"))
        assert(discussion.previousDiscussionUrl.startsWith("board_read.nhn?code=005930&nid=64102719"))

    }

    @Test
    fun testPreviousDiscussion() {
        val crawler = NaverStockDiscussionCrawler("005930", 1)

        val firstUrl = crawler.getStartDiscussionUrl()
        val discussion = crawler.getDiscussion(firstUrl)


        print(discussion)
        assert(discussion.previousDiscussionUrl.isEmpty())
        assert(discussion.nextDiscussionUrl.length > 10)

    }

    @Test
    fun testNoNextDiscussion() {
        val crawler = NaverStockDiscussionCrawler("005930", 1)
        val discussion = crawler.getDiscussion("/item/board_read.nhn?code=005930&nid=64104583&st=&sw=&page=46711")

        print(discussion)
        assert(discussion.nextDiscussionUrl.isEmpty())
        assert(discussion.previousDiscussionUrl.length > 10)

    }


}